### PR TITLE
Build Moose observation forms based on prototype

### PR DESCRIPTION
**Pull request by sdevalapurkar**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1135

## This PR contains the following changes

- Build Moose observation forms based on prototype

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)

## How Has This Been Tested?

- Locally

## Screenshots

image

image

### CODE REVIEW

File: app/src/features/surveys/view/SurveyObservations.tsx [hunk 1]
```diff
@@ -68,14 +70,27 @@ const SurveyObservations: React.FC<ISurveyObservationsProps> = (props) => {
...
+
   return (
     <>
       <Box mb={5} display="flex" alignItems="center" justifyContent="space-between">
         <Typography variant="h2">Observations</Typography>
         <Box>
           <Box display="flex" justifyContent="space-between">
             <Box mr={1}>
-              <Button variant="contained" color="primary" onClick={() => console.log('new block survey')}>
+              <Button variant="contained" color="primary" onClick={addNewObservation}>
```
Comment (anissa-agahchen) [inline app/src/features/surveys/view/SurveyObservations.tsx:93]: nice!

File: app/src/features/observations/components/BlockObservationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,750 @@
...
+import React, { useState } from 'react';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import { DATE_LIMIT } from 'constants/dateFormats';
+import { HotTable } from '@handsontable/react';
+import Handsontable from 'handsontable';
```
Comment (anissa-agahchen) [inline app/src/features/observations/components/BlockObservationForm.tsx:14]: You have a failing test here .. not sure what's causing it to fail on the local instance, but not in Openshift 
Screen Shot 2021-06-14 at 10 54 27 PM

Comment (sdevalapurkar) [reply app/src/features/observations/components/BlockObservationForm.tsx:14]: the tests arent failing for me and also arent failing on the CI checks...

Comment (NickPhura) [reply app/src/features/observations/components/BlockObservationForm.tsx:14]: `Cannot find module ...` type errors are likely a result of needing to run `npm install`

**Review by anissa-agahchen — APPROVED** [review]

**Pull request merged**